ClientRuntimes/CSharp/ClientRuntime.Azure/Authentication/ActiveDirectoryUserTokenProvider.cs

```diff
@@ -0,0 +1,195 @@
+            if (clientRedirectUri == null)
+            {
+                throw new ArgumentNullException("clientRedirectUri");
+            }
+
+            this._clientId = clientId;
+            this._tokenAudience = environment.TokenAudience.ToString();
+            string authority = environment.AuthenticationEndpoint + domain;
+            try
+            {
+                this._authenticationContext = new AuthenticationContext(authority,
+                    environment.ValidateAuthority);
+                var authenticatioResult = this._authenticationContext.AcquireTokenAsync(
+                    environment.TokenAudience.ToString(), clientId, clientRedirectUri,
+                    GetPlatformParameters(), UserIdentifier.AnyUser,
+                    EnableEbdMagicCookie).Result;
+                Initialize(authenticatioResult);
+            }
+            catch (AdalException authenticationException)
+            {
+                throw new AuthenticationException(Resources.ErrorAcquiringToken, authenticationException);
+            }
+        }
+
+        /// <summary>
+        /// Create a token provider using Active Directory user credentials (UPN). 
+        /// Authentication occurs using the given username and password, with no user prompt.
+        /// </summary>
+        /// <param name="clientId">The client id for thsi application.</param>
+        /// <param name="domain">The domain or tenant id contianing the resources to manage.</param>
+        /// <param name="username">The username to use for authentication.</param>
+        /// <param name="password">The secret password associated with thsi user.</param>
+        /// <param name="environment">The azure environment to manage resources in.</param>
+        public ActiveDirectoryUserTokenProvider(string clientId, string domain, string username, string password, AzureEnvironment environment) :
+            this(clientId, domain, username, password, environment, null)
+        {
+        }
+
+        /// <summary>
+        /// Create a token provider using Active Directory user credentials (UPN). 
+        /// Authentication occurs using the given username and password, with no user prompt.
+        /// </summary>
+        /// <param name="clientId">The client id for thsi application.</param>
+        /// <param name="domain">The domain or tenant id contianing the resources to manage.</param>
+        /// <param name="username">The username to use for authentication.</param>
+        /// <param name="password">The secret password associated with thsi user.</param>
+        /// <param name="environment">The azure environment to manage resources in.</param>
+        /// <param name="cache">The token cache to use during authentication.</param>
+        internal ActiveDirectoryUserTokenProvider(string clientId, string domain, string username, string password, AzureEnvironment environment, TokenCache cache)
```

Comment:
since the class is internal already, we can just set "public" if there are other types in the assemblies using it, or just "private"